src/main.rs

```diff
@@ -392,9 +397,22 @@ async fn attest(
         not_after: Time::try_from(end).or(Err(StatusCode::INTERNAL_SERVER_ERROR))?,
     };
 
+    // Check for correct mime type.
+    let (one_cert, cert_vec) = match ct.to_string().as_ref() {
+        PKCS10 => {
+            let cr = CertReq::from_der(body.as_ref()).or(Err(StatusCode::BAD_REQUEST))?;
+            (true, vec![cr])
+        }
+        PKCS10_BUNDLE => {
+            let crs =
+                Vec::<CertReq<'_>>::from_der(body.as_ref()).or(Err(StatusCode::BAD_REQUEST))?;
+            (false, crs)
+        }
+        _ => return Err(StatusCode::BAD_REQUEST),
+    };
```

Comment:
I don't like this `one_cert` pattern. Also, don't call it `cert_vec`. Since this isn't `Vec<Certificate>`, the name could confuse. This looks cleaner to me:

```suggestion
    // Check for correct mime type.
    let reqs = match ct.to_string().as_ref() {
        PKCS10 => vec![CertReq::from_der(body.as_ref()).or(Err(StatusCode::BAD_REQUEST))?],
        BUNDLE => Vec::from_der(body.as_ref()).or(Err(StatusCode::BAD_REQUEST))?,
        _ => return Err(StatusCode::BAD_REQUEST),
    };
```

Note that I don't think you need `Vec::<CertReq<'_>>` since the return value from the first branch of the match statement should help the compiler infer the type in the second branch.

Reply:
Done